### ISSUE TITLE
Witaj

### ISSUE THREAD

**Issue by Dreed420** [Aifryz/parkside-battery-hack#1]
Widzę że ogarnąłeś temat ładowarki - ja mam problem z baterią, wymieniłem koledze ogniwa na większe - bateria podziałała dwa dni i po podłączeniu do ładowarki mruga czerwona dioda (sprawdziłem ogniwa są o równym napięciu) - inne baterie ładują się prawidłowo.
Tyle co umiałem sprawdziłem i zauważyłem że jest problem z sygnałem ID
Czy mógłbym Cie prosić o soft do baterii którym mógłbym oszukać ładowarkę? 
Nie chciałbym przerabiać ładowarki (bo kolega ma ich parę) dla jednej baterii a sama elektronikę ciężko znaleźć - a widząc że masz wszystko rozgryzione myślę że to nie będzie problem dla Ciebie :))

Z góry dziękuje i czekam na odpowiedź
PS. Jeżeli potrzeba jakiś pomiarów aby usprawnić program to służę pomocą

**Comment by Dreed420** [comment]
Ładowarka to PLG 20 B1
Bateria PAP 20 A1

Szukam softu czym mógłbym zemulować sygnał ID bo padł układ LGT8P22A którego nie da się nigdzie zakupić

**Comment by Aifryz** [comment]
Wszystko rozgryzione to dużo za dużo powiedziane :) 
Teoretycznie ten program na arduino powinien spowodować że ładowarka rzeczywiście poda napięcie na baterie.
Ale wydaje mi się że układ LGT8P22A jest też częścią BMSa w samym akumulatorze i wyłącza ładowanie po stronie baterii. Nie wiem czy dałbym radę coś tu zdziałać bez analizowania całego akumulatora.

Próbowałeś ładować tą baterię zasilaczem z regulowanym limitem prądu wyjściowego?

**Comment by Dreed420** [comment]
tak zasilaczem serwisowym ładuje się bez problemu, poszedłem nawet dalej - 
wgrałem twój program do arduino, pod pin D9 podpiąłem pin ID w baterii oraz mase do GND i ładowarka ruszyła - lecz nie kończy ładowania.
Nie umieściłeś żadnego pinout'u więc też trochę idę po omacku ;)

Wczoraj dotarł do mnie analizator stanów logiczny, czekam na sprawną baterię od kolegi i będę mógł 'zgrać' cały przebieg. 

ps. dysponujesz jakims komunikatorem typu discord?

**Comment by Aifryz** [comment]
Racja, nie wrzuciłem jeszcze schematów, ale zgadłeś z tym pinem D9 pod pin ID.

Co do samego kończenia ładowania - w moim przypadku też właściwie nie było dobrej detekcji końca ładowania. jak się naładowała to zaczęły dwie diody razem migać (wg. instrukcji - błąd baterii).  Nigdy nie miałem potrzeby aby sprawdzić jak wygląda komunikacja na końcu ładowania, więc tego nie robiłem.

Być może w tym tygodniu, albo w weekend będę miał chwilę czasu to postaram się wymyślić jakiś sposób na zmierzenie tego.
Z takim zwykłym mierzeniem przy użyciu analizatora może być problem bo trzeba wychwycić moment kiedy bateria przestaje się ładować, a to może być po paru godzinach.

Discorda mam, ale używam go trochę jak maila, wyłączone powiadomienia i sprawdzam raz na dzień.

**Comment by Dreed420** [comment]
>Z takim zwykłym mierzeniem przy użyciu analizatora może być problem bo trzeba wychwycić moment kiedy bateria przestaje się ładować, a to może być po paru godzinach.


a włożenie naładowanej baterii i sprawdzenie jak wygląda 'finisz' ? :)

**Comment by Aifryz** [comment]
Nareszcie miałem okazję zobaczyć jak wyglądają sygnały podczas całego ładowania. Sygnał ID zmienia się kilkaset msek przed końcem ładowania. Czyli wygląda jakby to rzeczywiście bateria decydowała kiedy ma się odłączyć. Z tym nie uda mi się mi wiele zdziałać. 

Jedyny pomysł jak to zrobić to byłoby mierzenie prądu który płynie przez rezystor pomiarowy na plytce i odcinanie sygnału jak prąd spadnie poniżej jakiegoś progu. Tylko to wymagałoby modyfikacji ładowarki.

Samo to, że ładowarka nie przestaję ładować może być spowodowane też jakimś małym zwarciem w bateri (kilkadziesiąt-kilkaset ohm). Wtedy rzeczywiście ładowarka ciągle by musiała podawać prąd.
Można to sprawdzić mierząc prąd płynący do baterii.
